tests/logging/test_opentracing.py

```diff
@@ -277,3 +277,39 @@ async def fixture_async_func() -> str:
             [span.operation_name for span in self._reporter.get_spans()],
             ["fixture_async_func"],
         )
+
+    def test_trace_decorator_awaitable_return(self) -> None:
+        """
+        Test whether we can use `@trace_with_opname` (`@trace`) and `@tag_args`
+        with functions that return an awaitable (e.g. a coroutine)
+        """
+        reactor = MemoryReactorClock()
+
+        with LoggingContext("root context"):
+            # Something we can return without `await` to get a coroutine
+            async def fixture_async_func() -> str:
+                return "foo"
+
+            # The actual kind of function we want to test that returns an awaitable
+            @trace_with_opname("fixture_awaitable_return_func", tracer=self._tracer)
+            @tag_args
+            def fixture_awaitable_return_func() -> Awaitable[str]:
+                return fixture_async_func()
+
+            # Something we can run with `defer.ensureDeferred(runner())` and pump the
+            # whole async tasks through to completion.
+            async def runner() -> str:
+                return await fixture_awaitable_return_func()
+
+            d1 = defer.ensureDeferred(runner())
```

Comment:
Better way to do this without the layers?

Reply:
I'm not really sure what `runner` is doing -- can you remove that later and just do:

```python
d1 = defer.ensureDeferred(fixture_awaitable_return_func)
```

I'm fairly certain `ensureDeferred` works with `Awaitable` as well as `coroutine` and `Deferred`.

Reply:
Perfect! One less layer in all of this!

Reply:
> I'm fairly certain `ensureDeferred` works with `Awaitable` as well as `coroutine` and `Deferred`.

hmmm, actually, this sort of usage results in this type check lint:

```
tests/logging/test_opentracing.py:299: error: Argument 1 to "ensureDeferred" has incompatible type "Awaitable[str]"; expected "Union[Coroutine[Deferred[<nothing>], Any, <nothing>], Generator[Deferred[<nothing>], Any, <nothing>], Deferred[<nothing>]]"  [arg-type]
```

But in this case, we could just go back to the extra layer. For the other case in the other thread, we would have to figure out how to make any `Awaitable` into a `Deferred`